spec/models/ability_spec.rb

```diff
@@ -47,6 +47,20 @@
     end
   end
 
+  context "a notice viewer" do
+    subject { Ability.new(build(:user, :notice_viewer)) }
+
+    it_behaves_like 'a non-admin'
+
+    it "can't generate new notice token urls" do
+      expect(subject.can? :generate_permanent_notice_token_urls, Notice).to be false
```

Comment:
For which other roles should we be applying this check?

Reply:
Ok, got it, will update